src/main.rs

```diff
@@ -26,19 +35,30 @@ fn main() -> Result<()> {
         ColorChoice::Auto,
     )?;
 
+    // Step 3: get the modules.
+    let mut modules = get_modules()?;
+
+    // Step 4: dispatch the command.
     let command = cli.get_subcommand_mut()?;
     match command.name() {
         "collect" => {
```

Comment:
Inspired by the great improvements of this RFC and by the ability to use associated types on Boxed traits that I learnt on https://github.com/retis-org/retis/pull/102#discussion_r1165355400 I have added a potential improvement on top of this:
https://github.com/amorenoz/retis/commit/1b6eca3decb16858a45d944235249d0f0a7608c5

It allows us to move 100% of the command logic inside the command itself (in this case `collect/collect.rs`). I think we will appreciate it once we start adding more and more commands to the cli. WDYT?